src/main/kotlin/uk/gov/dluhc/printapi/rds/entity/Certificate.kt

```diff
@@ -94,28 +93,39 @@ class Certificate(
 
     fun addPrintRequest(newPrintRequest: PrintRequest): Certificate {
         printRequests += newPrintRequest
+        assignStatus()
         return this
     }
 
-    @PrePersist
-    fun prePersist() {
-        assignStatus()
+    fun getCurrentPrintRequest(): PrintRequest {
+        printRequests.sortByDescending { it.requestDateTime }
```

Comment:
Does `sortByDescending` affect (mutate) the collection, or does it return a new collection?
I think you might have to do this instead:
```
fun getCurrentPrintRequest(): PrintRequest =
        printRequests.sortByDescending { it.requestDateTime }.first()
```